mbedtls/src/x509/certificate.rs

```diff
@@ -1004,7 +1089,18 @@ cYp0bH/RcPTC0Z+ZaqSWMtfxRrk63MJQF9EXpDCdvQRcTMD9D85DJrMKn8aumq0M
             chain.push(c_int1.clone());
             chain.push(c_int2.clone());
 
-            Certificate::verify(&chain, &mut c_root, None).unwrap();
+            let verify_callback = |_crt: &Certificate, _depth: i32, verify_flags: &mut VerifyError| {
+                verify_flags.remove(VerifyError::CERT_EXPIRED);
+                Ok(())
+            };
+
+            let mut err_str = String::new();
+            let res = Certificate::verify_callback(&chain, &mut c_root, Some(&mut err_str), verify_callback);
+
+            match res {
+                Ok(()) => (),
+                Err(e) => assert!(false, "Failed to verify, error: {}, err_str: {}", e, err_str),
+            };
```

Comment:
Shouldn't we keep the original test (which uses `Certificate::verify()` instead of `Certificate::verify_callback()`) as well?